spikeinterface/core/unitsaggregationsorting.py

```diff
@@ -0,0 +1,93 @@
+from typing import List, Union
+
+import numpy as np
+
+from .basesorting import BaseSorting, BaseSortingSegment
+
+
+class UnitsAggregationSorting(BaseSorting):
+    """
+    Class that handles aggregating units from different sortings, e.g. from different channel groups.
+
+    Do not use this class directly but use `si.aggregate_units(...)`
+
+    """
+    def __init__(self, sorting_list):
+        unit_map = {}
+
+        # units are renamed from 0 to N-1
+        unit_ids = []
+        u_id = 0
+        for s_i, sorting in enumerate(sorting_list):
+            single_unit_ids = sorting.get_unit_ids()
+            for unit_id in single_unit_ids:
+                unit_ids.append(u_id)
+                unit_map[u_id] = {'sorting_id': s_i, 'unit_id': unit_id}
```

Comment:
Use `sorting_ind` please.

Reply:
if you check the [`get_unit_spike_train`](https://github.com/SpikeInterface/spikeinterface/blob/master/spikeinterface/core/basesorting.py#L49-L57), unit_ids are used, not indices. I think this might be wrong, but we first have to fix it there.

Reply:
@samuelgarcia should we fix this?

Reply:
No I think this is OK. I was thinking you were mixing unit_index and unit_id but you handle everything with id it is OK I think. I will check laterr.